### PR TITLE
cutouts.py

### PR DESCRIPTION
**Pull request by ceb8**
In this PR:
- Moved Tesscut functionality into cutouts.py (was tesscut.py)
- Made a service initialization function (`_set_service_params`) for `ServiceAPI` and moved the catalog specific parameters into `CatalogsClass`
- Moved tesscut function `_parse_coordinates` into utils and have catalogs and observations use it

### CODE REVIEW

File: astroquery/mast/collections.py [hunk 1]
```diff
@@ -241,20 +245,13 @@ def query_criteria_async(self, catalog, pagesize=None, page=None, **criteria):
...
-            raise InvalidQueryError("Only one of objectname and coordinates may be specified.")
-
-        if objectname:
-            coordinates = self.resolve_object(objectname)
-
-        if coordinates:
-            # Put coordinates and radius into consitant format
-            coordinates = commons.parse_coordinates(coordinates)
+        if objectname or coordinates:
+            coordinates = utils._parse_input_location(coordinates, objectname)
```
Comment (tomdonaldson) [inline astroquery/mast/collections.py:249]: I don't think a public utility method shouldn't be named with "_".

File: astroquery/mast/cutouts.py [hunk 2]
```diff
@@ -31,49 +31,48 @@
...
-    if not (objectname or coordinates):
-        raise InvalidQueryError("One of objectname and coordinates must be specified.")
+__all__ = ["TesscutClass", "Tesscut"]
 
-    if objectname:
-        obj_coord = resolve_object(objectname)
 
-    if coordinates:
-        obj_coord = commons.parse_coordinates(coordinates)
+def _parse_cutout_size(size):
```
Comment (tomdonaldson) [inline astroquery/mast/cutouts.py:41]: Even as a private method, could this have documentation on what sorts of values `size` is allowed to be?

File: astroquery/mast/collections.py [hunk 3]
```diff
@@ -40,6 +40,10 @@ def __init__(self):
 
         super().__init__()
 
+        services = {"panstarrs": {"path": "panstarrs/{data_release}/{table}.json",
+                                  "args": {"data_release": "dr2", "table": "mean"}}}
+        self._service_api_connection._set_service_params(services, "catalogs", True)
```
Comment (tomdonaldson) [inline astroquery/mast/collections.py:45]: The `ServiceAPI` class seems to have all private accessors (`_set_service_params()`, `_SERVICES`, etc.).  Are they meant to be public, and just need a name change, or are they actually private and not meant for the external access they're getting?

**Review by tomdonaldson — APPROVED** [review]
I like this refactoring a lot.  I've added some stylistic comments that could be addressed now or later, but should probably be addressed.

File: astroquery/mast/collections.py (hunk 1 above)
Comment (ceb8) [reply astroquery/mast/collections.py]: I wasn't thinking of it as a public method. Do you think it should be public?

**Comment by ceb8** [comment]
@tomdonaldson Your two questions about private vs  public methods brings up a larger question of how we are dividing public/private and what changes need to be made there. I am going to leave that work off this PR and include it in the list of things to address when doing the final refactor pass. I think it makes sense to look at all our classes and functions through this lens.

File: astroquery/mast/cutouts.py (hunk 2 above)
Comment (ceb8) [reply astroquery/mast/cutouts.py:41]: Good catch! Did not even notice I neglected a docstring there.

File: astroquery/mast/collections.py (hunk 1 above)
Comment (ceb8) [reply astroquery/mast/collections.py:249]: Will handle in future PR.

File: astroquery/mast/collections.py (hunk 3 above)
Comment (ceb8) [reply astroquery/mast/collections.py:45]: Will handle in future PR.

**Pull request merged**